test/integration/apiserver/flowcontrol/fs_condition_test.go

```diff
@@ -0,0 +1,151 @@
+	flowcontrolapply "k8s.io/client-go/applyconfigurations/flowcontrol/v1beta2"
+	clientset "k8s.io/client-go/kubernetes"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/klog/v2"
+)
+
+func TestConditionIsolation(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, true)()
+	// NOTE: disabling the feature should fail the test
+	_, loopbackConfig, closeFn := setup(t, 10, 10)
+	defer closeFn()
+
+	loopbackClient := clientset.NewForConfigOrDie(loopbackConfig)
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	ctx := context.Background()
+
+	fsOrig := fcboot.SuggestedFlowSchemas[0]
+	t.Logf("Testing Status Condition isolation in FlowSchema %q", fsOrig.Name)
+	fsClient := loopbackClient.FlowcontrolV1beta2().FlowSchemas()
+	var dangleOrig *flowcontrol.FlowSchemaCondition
+
+	wait.PollUntil(time.Second, func() (bool, error) {
+		fsGot, err := fsClient.Get(ctx, fsOrig.Name, metav1.GetOptions{})
+		if err != nil {
+			klog.Errorf("Failed to fetch FlowSchema %q: %v", fsOrig.Name, err)
+			return false, nil
+		}
+		dangleOrig = getCondition(fsGot.Status.Conditions, flowcontrol.FlowSchemaConditionDangling)
+		return dangleOrig != nil, nil
+	}, stopCh)
+
+	ssaType := flowcontrol.FlowSchemaConditionType("test-ssa")
+	patchSSA := flowcontrolapply.FlowSchema(fsOrig.Name).
+		WithStatus(flowcontrolapply.FlowSchemaStatus().
+			WithConditions(flowcontrolapply.FlowSchemaCondition().
+				WithType(ssaType).
+				WithStatus(flowcontrol.ConditionTrue).
+				WithReason("SSA test").
+				WithMessage("for testing").
+				WithLastTransitionTime(metav1.Now()),
+			))
+
+	postSSA, err := fsClient.ApplyStatus(ctx, patchSSA, metav1.ApplyOptions{FieldManager: "ssa-test"})
+	if err != nil {
+		t.Error(err)
+	}
+	danglePostSSA := getCondition(postSSA.Status.Conditions, flowcontrol.FlowSchemaConditionDangling)
+	if danglePostSSA == nil || danglePostSSA.Status != dangleOrig.Status {
+		t.Errorf("Bad dangle condition after SSA, the FS is now %s", fmtFS(t, postSSA))
+	}
+	ssaPostSSA := getCondition(postSSA.Status.Conditions, ssaType)
+	if ssaPostSSA == nil || ssaPostSSA.Status != flowcontrol.ConditionTrue {
+		t.Errorf("Bad SSA condition after SSA, the FS is now %s", fmtFS(t, postSSA))
+	}
+
+	smpType := flowcontrol.FlowSchemaConditionType("test-smp")
+	smpBytes, err := makeFlowSchemaConditionPatch(flowcontrol.FlowSchemaCondition{
+		Type:               smpType,
+		Status:             flowcontrol.ConditionFalse,
+		Reason:             "SMP test",
+		Message:            "for testing too",
+		LastTransitionTime: metav1.Now(),
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	postSMP, err := fsClient.Patch(ctx, fsOrig.Name, machinerytypes.StrategicMergePatchType, smpBytes,
+		metav1.PatchOptions{FieldManager: "smp-test"}, "status")
+	if err != nil {
+		t.Error(err)
+	}
+	if false /* disabled until patch annotations go into the API */ {
```

Comment:
Please link the issue here (otherwise it will not be clear for many people what patch annotations): https://github.com/kubernetes/kubernetes/issues/107574

Reply:
done